app/models/ebean/Report.java

```diff
@@ -194,6 +196,24 @@ public void setOrganization(final Organization value) {
         organization = value;
     }
 
+    /**
+     * Finds a {@link Report} when given a uuid.
+     *
+     * @param ebeanServer The {@code EbeanServer} instance to use.
+     * @param organization The parent organization for this report.
+     * @param reportId The report uuid to lookup.
+     * @return The report from the database.
+     */
+    public static Optional<Report> findByUUID(
+            final EbeanServer ebeanServer,
+            final models.ebean.Organization organization,
+            final UUID reportId) {
+        return ebeanServer.find(models.ebean.Report.class)
+                .where()
+                .eq("uuid", reportId)
+                .eq("organization.uuid", organization.getUuid())
+                .findOneOrEmpty();
```

Comment:
note: here you don't do the max rows.

Reply:
The difference is here I don't expect there to be more than one row with a given report/organization id, so it should throw.

The above code only specifies the report, but a report can have multiple executions. Since we want the latest one, we add the limit because otherwise we'd be grabbing them all (and triggering an exception in `findOneOrEmpty`.